### ISSUE TITLE
Check Final Leaf prediction in Maize Model

### ISSUE THREAD

**Issue by HamishBrownPFR** [APSIMInitiative/ApsimX#797]
I have refactored the Structure and Leaf classes handing of Final leaf number.  This was necessary to enable the setting of Final Leaf number from a manager (for potato) and has greatly simplified the code which was unnecessarily complex.  The changes have not had any effect on the wheat model but have had a small effect on the final leaf number predictions in the maize model which is surprising as they use practically the same method to predict Final leaf number.  I have merged the change in complete with changed stats on Maize model but the FLN predictions in Maize need to be reviewed to ensure the change is due to fixing a bug and not introducing one.  

Relates to changes made in pull request #796

**Comment by HamishBrownPFR** [comment]
I am more certain now this is a bug I have introduced as I am seeing some odd behavior in leaf number in the Gatton2009 trial which I fixed a couple of weeks ago.

**Comment by HamishBrownPFR** [comment]
No.  I have just rolled back to prior to the change that introduced the potential bug and the Wheat leaf number data I was looking at in the above comment has not changed.

**Comment by HamishBrownPFR** [comment]
I have traced the source of the differences and they have come about because the refactoring I did has fixed a minor bug that was in the LeafAppearancePhase.  Basically the highlighted property used to calculate FractionComplete was the incorrect value in the previous version of the model.  
![image](https://cloud.githubusercontent.com/assets/7295858/14740066/1d21439e-08e0-11e6-964b-6f1ac3b7c49e.png)
in the refactored version I have got rid of this property and just reference the function that provides this parameter dirrectly.  The Value for MainStemFinalNodeNumber changes from and estimate to an actual upon moving into the leaf appearance phase.  In the previous version it was using the estimated value and in the refactored version it is using the actual value which is correct.  This error only came up on the first day of the leaf appearance phase but was enough to influence Phenology.Stage which influenced height (which references that) which influenced Water use .....  I will now close this issue as I am satisified that the changes to the models statistics are the result of an improvement (through fixing a bug) and not the result of making a bug.  A seperate issue has arrisen here where the stats for final leaf number in the Australian validation loom far worse.  This is because the PO object has turned predicted final leaf numbers from doubles into integers which has impacted the stats.
